src/views/header/ConservationView.js

```diff
@@ -14,23 +18,72 @@ const ConservationView = view.extend({
     // we need to wait until stats gives us the ok
     //@listenTo @model, "reset",@render
     this.listenTo(this.g.stats,"reset", this.render);
+    
+    var opts = _.extend( {}, { 
+      fillColor: '#660', // [ '#660', '#ff0'],
+      strokeColor: '#330',
+      maxHeight: 20,
+      rectStyler: function(rect, data) { return rect }
+    }, this.g.conservationConfig );
+        
+    this.fillColor = opts.fillColor;
+    this.strokeColor = opts.strokeColor;
+    this.maxHeight = opts.maxHeight;
+    this.rectStyler = opts.rectStyler;
+    
     return this.manageEvents();
   },
-
+  
+  // returns a function that will decide a colour 
+  // based on the conservation data it is given
+  colorer: function(colorRange) {
+    var colorer = function() { return "none" };
+        
+    if( typeof colorRange === 'string' ) {
+      colorer = function() { return colorRange };
+    }
+    else if( Array.isArray( colorRange ) ) {
+      if ( colorRange.length != 2 ) {
+        console.error( "ERROR: colorRange array should have exactly two elements", colorRange );
+      }
+      
+      if ( !d3_scale ) {
```

Comment:
thanks for all the comments - all very useful

I've gone with your code to check whether d3 is available (ignore d3-scale)





On 12 May 2016 at 17:45, Sebastian Wilzbach <notifications@github.com>
wrote:

> In src/views/header/ConservationView.js
> <https://github.com/wilzbach/msa/pull/130#discussion_r63054556>:
>
> > -
> > +
> > +  // returns a function that will decide a colour
> > +  // based on the conservation data it is given
> > +  colorer: function(colorRange) {
> > +    var colorer = function() { return "none" };
> > +
> > +    if( typeof colorRange === 'string' ) {
> > +      colorer = function() { return colorRange };
> > +    }
> > +    else if( Array.isArray( colorRange ) ) {
> > +      if ( colorRange.length != 2 ) {
> > +        console.error( "ERROR: colorRange array should have exactly two elements", colorRange );
> > +      }
> > +
> > +      if ( !d3_scale ) {
>
> oh then it's probably better to check for typeof d3_scale == "undefined"
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/wilzbach/msa/pull/130/files/1c5231d9e1a0b9d3b16e2155df62ecfb1329874c#r63054556>
>



-- 
*Ian Sillitoe* | Senior Research Fellow
*CATH Manager* | www.cathdb.info
*University College London* | 627 Darwin, Gower Street, WC1E 6BT